src/hardware/vga_dac.cpp

```diff
@@ -60,32 +60,88 @@ Note:  Each read or write of this register will cycle through first the
 
 enum { DacRead, DacWrite };
 
-static void vga_dac_send_color(const uint8_t palette_idx, const uint8_t color_idx)
+static bool is_cga_color(Rgb666 color)
 {
-	const auto rgb666 = vga.dac.rgb[color_idx];
-
-	const auto r8 = rgb6_to_8_lut(rgb666.red);
-	const auto g8 = rgb6_to_8_lut(rgb666.green);
-	const auto b8 = rgb6_to_8_lut(rgb666.blue);
+	for (const auto cga_color : palette.cga16) {
+		if (cga_color == color) {
+			return true;
+		}
+	}
+	return false;
```

Comment:
Thanks for the tips @japsmits! These work great, so changing to the C+11 style. Unfortunately, we can't upgrade to C++20 yet AFAIK (because of the Raspberry Pi).